lt2opencorpora/convert.py

```diff
@@ -284,20 +288,37 @@ def __init__(self, fname, mapping):
                     ))
 
             self.add_lemma(current_lemma)
+        self.write_tree()
+        self.write_final()
 
-    def add_lemma(self, lemma):
+    def add_lemma(self, lemma: Lemma):
         if lemma is not None:
-            self.lemmas[lemma.lemma_signature] = lemma
-
-    def export_to_xml(self, fname):
-        root = ET.Element("dictionary", version="0.2", revision="1")
-        tree = ET.ElementTree(root)
-        root.append(self.tag_set.export_to_xml())
-        lemmata = ET.SubElement(root, "lemmata")
-
-        for i, lemma in enumerate(self.lemmas.values()):
-            lemma_xml = lemma.export_to_xml(i + 1)
+            lemma_xml = lemma.export_to_xml(self.counter)
             if lemma_xml is not None:
-                lemmata.append(lemma_xml)
-
-        tree.write(fname, encoding="utf-8")
+                self.counter += 1
+                self.lemmata.append(lemma_xml)
+
+    def write_tree(self):
+        self.tree.write('temp.xml', encoding="utf-8")
+        with open('temp.xml') as f:
+            file_content = f.read()
+            file_content = '\n' + file_content[9:-10]
+            with open('result.xml', 'a') as af:
+                af.write(file_content)
+        os.remove('{}/temp.xml'.format(os.path.abspath(os.getcwd())))
+        self.lemmata.clear()
+
+    def write_final(self):
+        # line where to insert result.xml
+        template_start = 440
```

Comment:
This shit is evil too. You change one line in the template and you are in trouble.

Once again, I'd like to hear what is the rationale behind that.

Reply:
Fix in https://github.com/dchaplinsky/LT2OpenCorpora/pull/26/commits/8327d8809623d529f604e087ec038697c20cffe5